### PR TITLE
spring-06

### PR DESCRIPTION
**Pull request by AndreiYazadzhi**

### CODE REVIEW

**Review by gigbat — APPROVED** [review]

File: src/main/java/com/hibernate/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                           ShoppingCartService shoppingCartService,
+                           UserService userService,
+                           DtoResponseMapper<OrderResponseDto, Order> responseMapper) {
+        this.orderService = orderService;
+        this.shoppingCartService = shoppingCartService;
+        this.userService = userService;
+        this.responseMapper = responseMapper;
+    }
+
+    @PutMapping("/complete")
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/controller/OrderController.java]: why PutMapping?

File: src/main/java/com/hibernate/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    public void completeOrder(OrderRequestDto dto) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/controller/OrderController.java]: see task description - how should the data be passed to this method?

File: src/main/java/com/hibernate/controller/ShoppingCartController.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+@RestController
+@RequestMapping("/shopping-carts")
+public class ShoppingCartController {
+    private final MovieSessionService movieSessionService;
+    private final ShoppingCartService shoppingCartService;
+    private final UserService userService;
+    private final DtoResponseMapper<ShoppingCartResponseDto, ShoppingCart> responseMapper;
+
+    public ShoppingCartController(MovieSessionService movieSessionService,
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/controller/ShoppingCartController.java:25]: let's either use `@Autowired` annotation above the constructor everywhere - or not use it at all

File: src/main/java/com/hibernate/model/dto/mapping/impl/request/UserRequestMapping.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package com.hibernate.model.dto.mapping.impl.request;
+
+import com.hibernate.model.User;
+import com.hibernate.model.dto.mapping.DtoRequestMapper;
+import com.hibernate.model.dto.request.UserRequestDto;
+
+public class UserRequestMapping implements DtoRequestMapper<UserRequestDto, User> {
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/model/dto/mapping/impl/request/UserRequestMapping.java]: do we need this?

File: src/main/java/com/hibernate/model/dto/mapping/impl/response/OrderResponseMapper.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import java.time.format.DateTimeFormatter;
+import java.util.stream.Collectors;
+
+public class OrderResponseMapper implements
+        DtoResponseMapper<OrderResponseDto, Order> {
+    @Override
+    public OrderResponseDto toDto(Order order) {
+        OrderResponseDto orderResponseDto = new OrderResponseDto();
+        orderResponseDto.setOrderDate(order.getOrderDate()
+                .format(DateTimeFormatter.ofPattern("dd.MM.yyyy HH:mm")));
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/model/dto/mapping/impl/response/OrderResponseMapper.java]: let's move pattern to a constant field

File: src/main/java/com/hibernate/model/dto/mapping/impl/response/UserResponseMapping.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
+package com.hibernate.model.dto.mapping.impl.response;
+
+import com.hibernate.model.User;
+import com.hibernate.model.dto.mapping.DtoResponseMapper;
+import com.hibernate.model.dto.response.UserResponseDto;
+
+public class UserResponseMapping implements DtoResponseMapper<UserResponseDto, User> {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/model/dto/mapping/impl/response/UserResponseMapping.java]: redundant empty line

File: src/main/java/com/hibernate/model/dto/response/UserResponseDto.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
+package com.hibernate.model.dto.response;
+
+public class UserResponseDto {
+    private String email;
+    private String password;
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/model/dto/response/UserResponseDto.java]: see checklist - we should not expose the secure data, but should return id in all ResponseDtos

File: src/main/java/com/hibernate/model/dto/response/ShoppingCartResponseDto.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
+package com.hibernate.model.dto.response;
+
+import java.util.List;
+
+public class ShoppingCartResponseDto {
+    private Long cartId;
+    private List<Long> ticketsId;
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/model/dto/response/ShoppingCartResponseDto.java]:
```suggestion
    private List<Long> ticketIds;
```

File: src/main/java/com/hibernate/model/dto/response/OrderResponseDto.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
+package com.hibernate.model.dto.response;
+
+import java.util.List;
+
+public class OrderResponseDto {
+    private Long id;
+    private List<Long> ticketsId;
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/model/dto/response/OrderResponseDto.java]: same

File: src/main/java/com/hibernate/model/dto/request/ShoppingCartRequestDto.java [hunk 10]
```diff
@@ -0,0 +1,24 @@
+package com.hibernate.model.dto.request;
+
+import java.util.List;
+
+public class ShoppingCartRequestDto {
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/model/dto/request/ShoppingCartRequestDto.java]: do we use this?

File: src/main/java/com/hibernate/model/dto/request/OrderRequestDto.java [hunk 11]
```diff
@@ -0,0 +1,24 @@
+package com.hibernate.model.dto.request;
+
+import java.util.List;
+
+public class OrderRequestDto {
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/model/dto/request/OrderRequestDto.java]: same - we don't need this

File: src/main/java/com/hibernate/model/dto/mapping/impl/response/UserResponseMapping.java [hunk 12]
```diff
@@ -0,0 +1,16 @@
+package com.hibernate.model.dto.mapping.impl.response;
+
+import com.hibernate.model.User;
+import com.hibernate.model.dto.mapping.DtoResponseMapper;
+import com.hibernate.model.dto.response.UserResponseDto;
+
+public class UserResponseMapping implements DtoResponseMapper<UserResponseDto, User> {
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/model/dto/mapping/impl/response/UserResponseMapping.java]:
```suggestion
public class UserResponseMapper implements DtoResponseMapper<UserResponseDto, User> {
```

File: src/main/java/com/hibernate/model/dto/response/ShoppingCartResponseDto.java [hunk 13]
```diff
@@ -0,0 +1,24 @@
+package com.hibernate.model.dto.response;
+
+import java.util.List;
+
+public class ShoppingCartResponseDto {
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/model/dto/response/ShoppingCartResponseDto.java:5]: where is the mapper for ShoppingCart?

File: src/main/java/com/hibernate/security/AuthenticationServiceImpl.java [hunk 14]
```diff
@@ -34,8 +33,8 @@ public User register(String email, String password) {
 
     @Override
     public User login(String email, String password) {
-        Optional<User> user = userService.getByEmail(email);
-        if (user.isPresent() && user.get().getPassword()
+        Optional<User> user = Optional.of(userService.getByEmail(email));
+        if (user.get().getPassword()
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/security/AuthenticationServiceImpl.java]: we should not change this logic - this does not make sense at the moment

File: src/main/java/com/hibernate/service/UserService.java [hunk 15]
```diff
@@ -2,12 +2,13 @@
 
 import com.hibernate.model.User;
 import java.util.List;
-import java.util.Optional;
 
 public interface UserService {
     User add(User user);
 
-    Optional<User> getByEmail(String email);
+    User getByEmail(String email);
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/service/UserService.java]: same

File: src/main/java/com/hibernate/service/impl/UserServiceImpl.java [hunk 16]
```diff
@@ -28,8 +27,13 @@ public User add(User user) {
     }
 
     @Override
-    public Optional<User> getByEmail(String email) {
-        return userDao.getByEmail(email);
+    public User getByEmail(String email) {
+        return userDao.getByEmail(email).get();
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/service/impl/UserServiceImpl.java:33]: same

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/hibernate/controller/UserController.java [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+    private final DtoResponseMapper<UserResponseDto, User> responseMapper;
+    private final UserService userService;
+
+    public UserController(DtoResponseMapper<UserResponseDto, User> responseMapper,
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/controller/UserController.java:20]: add annotation

File: src/main/java/com/hibernate/controller/UserController.java [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+                          UserService userService) {
+        this.responseMapper = responseMapper;
+        this.userService = userService;
+    }
+
+    @GetMapping
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/controller/UserController.java]: what about mapping here?

File: src/main/java/com/hibernate/model/dto/mapping/impl/response/OrderResponseMapper.java [hunk 19]
```diff
@@ -0,0 +1,21 @@
...
+        DtoResponseMapper<OrderResponseDto, Order> {
+    private static final String PATTERN = "dd.MM.yyyy HH:mm";
+
+    @Override
+    public OrderResponseDto toDto(Order order) {
+        OrderResponseDto orderResponseDto = new OrderResponseDto();
+        orderResponseDto.setOrderDate(order.getOrderDate()
+                .format(DateTimeFormatter.ofPattern(PATTERN)));
+        orderResponseDto.setId(order.getId());
+        orderResponseDto.setUserEmail(order.getUser().getEmail());
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/model/dto/mapping/impl/response/OrderResponseMapper.java:22]: what about ticketIds?

File: src/main/java/com/hibernate/model/dto/response/OrderResponseDto.java [hunk 20]
```diff
@@ -0,0 +1,31 @@
+package com.hibernate.model.dto.response;
+
+public class OrderResponseDto {
+    private Long id;
+    private String orderDate;
+    private String userEmail;
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/model/dto/response/OrderResponseDto.java:8]:
```suggestion
    private String userEmail;
    private List<Long> ticketIds;
```

File: src/main/java/com/hibernate/model/dto/response/MovieResponseDto.java [hunk 21]
```diff
@@ -1,9 +1,18 @@
 package com.hibernate.model.dto.response;
 
 public class MovieResponseDto {
+    private Long id;
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/model/dto/response/MovieResponseDto.java:4]: let's set it in mapper

File: src/main/java/com/hibernate/model/dto/response/MovieSessionResponseDto.java [hunk 22]
```diff
@@ -1,11 +1,20 @@
 package com.hibernate.model.dto.response;
 
 public class MovieSessionResponseDto {
+    private Long id;
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/model/dto/response/MovieSessionResponseDto.java:4]: same here

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**